app/handlers/block_factory.py

```diff
@@ -0,0 +1,48 @@
+from app.handlers.blocks.content import Content
+from app.handlers.blocks.question import Question
+from app.handlers.blocks.list_collector import ListCollector
+from app.handlers.blocks.list_add_question import ListAddQuestion
+from app.handlers.blocks.list_edit_question import ListEditQuestion
+from app.handlers.blocks.list_remove_question import ListRemoveQuestion
+from app.handlers.blocks.primary_person_list_collector import PrimaryPersonListCollector
+from app.handlers.blocks.primary_person_question import PrimaryPersonQuestion
+from app.handlers.blocks.relationship_collector import RelationshipCollector
+from app.handlers.blocks.summary import Summary
+from app.handlers.blocks.section_summary import SectionSummary
+from app.handlers.blocks.calculated_summary import CalculatedSummary
+from app.questionnaire.location import InvalidLocationException
+
+BLOCK_MAPPINGS = {
+    'Question': Question,
+    'ConfirmationQuestion': Question,
+    'ListCollector': ListCollector,
+    'ListAddQuestion': ListAddQuestion,
+    'ListEditQuestion': ListEditQuestion,
+    'ListRemoveQuestion': ListRemoveQuestion,
+    'PrimaryPersonListCollector': PrimaryPersonListCollector,
+    'PrimaryPersonListAddOrEditQuestion': PrimaryPersonQuestion,
+    'RelationshipCollector': RelationshipCollector,
+    'Introduction': Content,
+    'Interstitial': Content,
+    'Confirmation': Content,
+    'Summary': Summary,
+    'SectionSummary': SectionSummary,
+    'CalculatedSummary': CalculatedSummary,
+}
+
+
+class BlockFactory:
+    @staticmethod
+    def get_handler(schema, location, questionnaire_store, language):
+        block = schema.get_block(location.block_id)
+        if not block:
+            raise InvalidLocationException(
+                f'block id {location.block_id} is not valid for this schema'
+            )
+
+        block_type = block['type']
+        block_class = BLOCK_MAPPINGS.get(block_type)
+        if not block_class:
```

Comment:
Do we need this?. If it is a valid block, then it should be a valid block type since it goes through validator?

Reply:
It will be useful when developing and adding new block types, I would have thought it makes sense to keep it.

Reply:
I think we should have this as a block type can be added without an implementation in runner. I will change it to an `ValueError` rather than `InvalidLocationException` as it shouldn't happen.